### ISSUE TITLE
In theory the block can even change between calls in one test case (inside cheatnet). We don't get the block number to later use it at any point during the execution -  we always use the `block_id`. A more consistent solution would be to get the block number in `run_from_test_case` using the `block_id` and then always use the specific block number, but it is to do in another PR

### ISSUE THREAD

**Issue by piotmag769** [foundry-rs/starknet-foundry#795]
In theory the block can even change between calls in one test case (inside cheatnet). We don't get the block number to later use it at any point during the execution -  we always use the `block_id`. A more consistent solution would be to get the block number in `run_from_test_case` using the `block_id` and then always use the specific block number, but it is to do in another PR

_Originally posted by @piotmag769 in https://github.com/foundry-rs/starknet-foundry/pull/783#discussion_r1344025270_

**Comment by piotmag769** [comment]
Useful code:
```rust
fn get_block_message(fork_params: Option<&(Url, BlockId)>) -> Result<String> {
    let result = if let Some((url, block_id)) = fork_params {
        match block_id {
            BlockId::Tag(_) => {
                let client = JsonRpcClient::new(HttpTransport::new(url.clone()));
                let runtime = Runtime::new().expect("Could not instantiate Runtime");
                match runtime.block_on(client.get_block_with_txs(block_id))? {
                    Block(BlockWithTxs {
                        block_number,
                        block_hash,
                        ..
                    }) => format!("\nBlock number: {block_number}, block hash: {block_hash}"),
                    PendingBlock(PendingBlockWithTxs { parent_hash, .. }) => {
                        format!("\nBlock parent hash: {parent_hash}")
                    }
                }
            }
            _ => String::new(),
        }
    } else {
        String::new()
    };

    Ok(result)
}
```

**Comment by piotmag769** [comment]
This is blocked by #775, wouldn't start it before that (major code changes on the side of forge)
